### PR TITLE
deepbluecap_contrib_traces

### PR DESCRIPTION
**Pull request by briot**
The output was no longer displayed in red, and instead was showing
the location, as in:
   [HANDLE] file.adb:100Error message
(not the missing space, too). The expected output is
   [HANDLE] Error message
where error message is displayed in red

### CODE REVIEW

File: src/gnatcoll-traces.adb [hunk 1]
```diff
@@ -1781,89 +1829,144 @@ package body GNATCOLL.Traces is
...
-                                    Relative_Path_To, Append);
+                        if Stream_Start /= 0 then
+                           declare
+                              Save   : Integer := Stream_Start + 1;
+                              Append : Boolean;
+                           begin
+                              if Stream_Start + 1 <= Stream_End
+                                 and then Config (Stream_Start + 1) = '>'
+                              then
+                                 Append := True;
```
Comment (anisimkov) [inline src/gnatcoll-traces.adb]: Append can be uninitialized

**Comment by briot** [comment]
> @anisimkov commented on this pull request.
> Append can be uninitialized


You are good at this game :-) !
fixed and amended

Manu

**Comment by anisimkov** [comment]
It was mostly gnatcoll regtests win.

**Comment by briot** [comment]
> It was mostly gnatcoll regtests win.

Interesting. Was it on the gnatcoll testsuite itself ? I did not see any error
myself. Or on another testsuite (infosys) ?

**Comment by anisimkov** [comment]
Just in gnatcoll testsuite. I guess uninitialized value can be different in different situations. I tried on 32 bit linux.

**Comment by anisimkov** [comment]
Just to let you know. New trace conf parser unable to skip spaces after equal sign.
Previous algorithm was able to do it.
As a result it is not recognizing turning off options like this
SQL = no
I was trying to fix algorithm itself but was lost in too complicated code. 
I'm trying to keep this patch and prepared fast fix pre-processing the data before parsing removing the spaces around equal sign with help of GNAT.Regpat package.

**Comment by briot** [comment]
I will take a look. The new parser is actually easier than that old one, will amend the patch

**Comment by anisimkov** [comment]
112fa722e src/gnatcoll-traces.adb 75..86

```
Brown_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
   ((Style => GNATCOLL.Terminal.Reset_All,
     Fg    => GNATCOLL.Terminal.Yellow,
     Bg    => GNATCOLL.Terminal.Unchanged));
Cyan_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
   ((Style => GNATCOLL.Terminal.Unchanged,
     Fg    => GNATCOLL.Terminal.Cyan,
     Bg    => GNATCOLL.Terminal.Unchanged));
Purple_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
   ((Style => GNATCOLL.Terminal.Reset_All,
     Fg    => GNATCOLL.Terminal.Magenta,
     Bg    => GNATCOLL.Terminal.Unchanged));
```

Why Brown and Purple Style Reset_All,
but Cyan Unchanged ?

**Comment by briot** [comment]
> Brown_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
>    ((Style => GNATCOLL.Terminal.Reset_All,
>      Fg    => GNATCOLL.Terminal.Yellow,
>      Bg    => GNATCOLL.Terminal.Unchanged));
> Cyan_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
>    ((Style => GNATCOLL.Terminal.Unchanged,
>      Fg    => GNATCOLL.Terminal.Cyan,
>      Bg    => GNATCOLL.Terminal.Unchanged));
> Purple_Fg  : constant String := GNATCOLL.Terminal.Get_ANSI_Sequence
>    ((Style => GNATCOLL.Terminal.Reset_All,
>      Fg    => GNATCOLL.Terminal.Magenta,
>      Bg    => GNATCOLL.Terminal.Unchanged));
> 
> Why Brown and Purple Style Reset_All,
> but Cyan Unchanged ?

Minor speed improvement: in the way they are used (and that was the case before
already), the foreground after Cyan is already reset, so we do not need to send
extra ANSI sequences to do that. This saves a few bytes in the output.

**Comment by anisimkov** [comment]
I had to revert this merge too.

**Comment by anisimkov** [comment]
It was not my decision.

**Comment by anisimkov** [comment]
windows not colorised in this test

**Comment by anisimkov** [comment]
gnatcoll/testsuite/traces/colors> obj/colors conf3.cfg 
[MODULE] Message 1
[RED] Message 2
[BLUE] Message 3
[DIM] Message 4
[RED] Message 5
[MODULE.ERROR] Message 6
[BLUE.ERROR] Message 7
[DIM.ERROR] Message 8

**Comment by briot** [comment]
> windows not colorised in this test

I do not have a compiler for Windows, so can’t help here much, sorry.

At first sight, it is expected.

The configuration file is forcing color output, but the only way that
GNATCOLL can do color output is by sending so-called ANSI sequences.

On Windows, this requires special tools to be installed (see gnatcoll-terminal.ads).
If this tool is not installed, GNATCOLL cannot do color output, so it is disabled
automatically.

(gnatcoll-terminal has support for using the proper windows API for colored
output, but it is not usable from GNATCOLL.Traces which needs to create a
string).

Perhaps gnatcoll-terminal.ads needs to be enhanced to better discover whether
ansi sequences are supported in the current terminal or no.

(code is in gnatcoll-traces.adb:558)

**Comment by anisimkov** [comment]
I'm trying to fix regressions on windows and see some strange behavior related to windows regression on linux. 
gnatcoll/testsuite/traces/colors> obj/colors conf3.cfg 
and
gnatcoll/testsuite/traces/colors> obj/colors conf3.cfg | cat
a bit different, the line is colored in first case but not colored in second.
Can you explain why it is happened ?
I was trying to fix related regression on windows. I set ANSICON environment there and have output with ANSI codes, but the DIM line is with escape codes, which is regression.

**Comment by anisimkov** [comment]
I guess I understand. It is because DIM trace handle output directed to stdout and detected when redirected to the file.

**Pull request merged**